src/secure_tunnel_watcher.py

```diff
@@ -16,6 +16,32 @@
 
 LOCK_FILE_PATH = "/app/lock/"
 
+def extract_http_proxy_url(url):
+    # Splitting URL into different parts
+    parts = url.split('@')
+    # Extracting user, password, host, and port
+    if len(parts) == 2:  # Userinfo is present
+        userinfo, host_port = parts
+        user_pass = userinfo.split(':')
+        if len(user_pass) == 2:  # Both user and password are present
+            user, password = user_pass
+        else:
+            user = user_pass[0]
+            password = None
+    else:
+        user = None
+        password = None
+        host_port = parts[0]
+
+    host_port_split = host_port.split(':')
+    host = host_port_split[0]
+    if len(host_port_split) == 2:
+        port = host_port_split[1]
+    else:
+        port = None
+
+    return user, password, host, port
```

Comment:
Please add tests to verify the different parsing behaviours with/without user-password-host-port combinations.